packages/protocol/contracts/L2/TaikoL2.sol

```diff
@@ -173,7 +175,11 @@ contract TaikoL2 is EssentialContract, TaikoL2Signer, ICrossChainSync {
 
     /// @notice Returns EIP1559 related configurations
     function getConfig() public pure virtual returns (Config memory config) {
-        config.gasTargetPerL1Block = 15 * 1e6 * 10; // 10x Ethereum gas target
+        // 4x Ethereum gas target, if we assume most of the time, L2 block time
+        // is 3s, and each block is full (gasUsed is 15_000_000), then its
+        // ~60_000_000, if the  network is congester than that, the base fee
+        // will increase.
+        config.gasTargetPerL1Block = 15 * 1e6 * 4;
```

Comment:
Initially the network won't be used much, so the L1 capacity may higher than our expected block size. So maybe on top of this calculation, divide it by a factor of 8? Then gradually we change the factor to 1?

Reply:
Otherwise, we'll always see 0 basefee

Reply:
For testnet, what we observed is that the network's usage is just simply highly related to community campaigns, but yeah for mainnet, that factor makes sense to me.

Reply:
So should we divided this number by a factor? Our testnet volume won't be larger than Ethereum mainnet, in terms of gas.

Reply:
Did some simulation. 

About "dynamically changing" values in the protocol:
Keeping track of the average gas / L1 variable and then adjust the behavior (all 3 values: gasExcess, target, quotient) to “show” a good look (like 1-10gwei baseFee) can technically be feasible but very challenging, computation heavy and is fee market manipulation, so I would not go this way.

 Also very important to note that the 3 values (gasExcess, target, quotient) are heavily interconnected ! If we change the target (+quotient) the gasExcess needs to change as well ! Otherwise the 1559 math will not work.
 
 When **60mil** and adjustment quotient is **8**, gas excess (calculated) is **19542030010** .
 
 THis is the simulation with the parameters (you can see in the labels). 
 
**Above target:**
![block (if gas](https://github.com/taikoxyz/taiko-mono/assets/51912515/8c9752d0-dc63-4cc3-8b68-77ac3eff4649)
**Below target:**
![block (if gas](https://github.com/taikoxyz/taiko-mono/assets/51912515/9df9e654-5b03-4392-887b-668ba41b40a1)
**Around target:**
![1698392033_target_traffic](https://github.com/taikoxyz/taiko-mono/assets/51912515/31f4d09d-bbc4-49bd-aff7-eab36b34f9f1)


  When we run the same simulation with a very low (wrong gas excess ) value: 1 it looks like this:  ( See y-axis with 1e-9) 
  
![block (if gas](https://github.com/taikoxyz/taiko-mono/assets/51912515/e28c0351-545f-4525-8ebf-fa8bd2851939)
![below target)](https://github.com/taikoxyz/taiko-mono/assets/51912515/045e015c-9ad6-48ce-a338-a97508ec8f8e)
![1698392540_target_traffic](https://github.com/taikoxyz/taiko-mono/assets/51912515/c18ba9cb-3eef-4d79-a50d-3febb746645f)


  When we use a huge one: 19542030010 (original good) * 10  (See y-axis with 1e41)
  
![1698392740_above_average_traffic](https://github.com/taikoxyz/taiko-mono/assets/51912515/1e17dd80-39bd-4459-97b3-7c7d23c946a8)
![below target)](https://github.com/taikoxyz/taiko-mono/assets/51912515/75c6dbb0-9309-4448-816d-a1aae2c88d88)
![Basefee chart](https://github.com/taikoxyz/taiko-mono/assets/51912515/7ebbc0c2-341e-4c20-81c2-9a8237a2aa7b)
 
 So this is my proposal:
**1.** Do not set the init gasExcess to wrong value :) (What would be the initial correct value, see point  
**2.** What is the most important (and currently lacking in TaikoL2) to have a setter to modify these 3 variables all together otherwise things would look more ugly.
**3.** Monitor A5 - what was the traffic in the first 5 days, I mean average gasUsed / L1 block, and use this as our base target, and calculate the gasExcess accordingly. It is a good starting point, and when things will differ highly, just use the function in point 2.

Reply:
@adaki2004 could you please add the setter function? We can still use a not-so-accurate initial gasExcess then use the function to turn the parameter later on. I don't think picking the initial value right for gasExcess shall be important.

Reply:
see https://github.com/taikoxyz/taiko-mono/pull/15052